routers/repo/issue_timetrack.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+	"net/http"
+	"time"
+	"fmt"
+)
+
+// AddTimeManual tracks time manually
+func AddTimeManual(c *context.Context) {
+
+	h, err := parseTimeTrackingWithDuration(c.Req.PostForm.Get("hours"), "h")
+	if err != nil {
+		fmt.Println("hours is not numeric", err)
+		c.Handle(http.StatusBadRequest, "hours is not numeric", err)
+		return
+	}
+
+	m, err := parseTimeTrackingWithDuration(c.Req.PostForm.Get("minutes"), "m")
+	if err != nil {
+		fmt.Println("minutes is not numeric", err)
+		c.Handle(http.StatusBadRequest, "minutes is not numeric", err)
+		return
+	}
+
+	s, err := parseTimeTrackingWithDuration(c.Req.PostForm.Get("seconds"), "s")
+	if err != nil {
+		fmt.Println("seconds is not numeric", err)
+		c.Handle(http.StatusBadRequest, "seconds is not numeric", err)
+		return
+	}
+
+	totalInSeconds := h.Seconds() + m.Seconds() + s.Seconds()
```

Comment:
Keep time.Duration until the end like `totalDurantion := h + m + s`. You can still compare totalDuration to 0 and do `int64(totalDurantion.Seconds())` at L53